tests/Struct/FindologicProductTest.php

```diff
@@ -186,6 +187,95 @@ public function testNoProductCategories(): void
         );
     }
 
+    /**
+     * @dataProvider categoryAndCatUrlWithIntegrationTypeProvider
+     */
+    public function testOnlyUniqueCategoriesExported(
+        ?string $integrationType,
+        array $categories,
+        array $expectedCatUrls
+    ): void {
+        foreach ($categories as $key => $category) {
+            $navigationCategoryId = $this->salesChannelContext->getSalesChannel()->getNavigationCategoryId();
+            $categories[$key]['parentId'] = $navigationCategoryId;
+        }
+
+        $productEntity = $this->createTestProduct(['categories' => $categories]);
+        $config = $this->getMockedConfig($integrationType);
+        $findologicProductFactory = new FindologicProductFactory();
+        $findologicProduct = $findologicProductFactory->buildInstance(
+            $productEntity,
+            $this->router,
+            $this->getContainer(),
+            $this->shopkey,
+            [],
+            new XMLItem('123'),
+            $config
+        );
+        $categoriesToBeChecked = [];
+        $isUnique = true;
+        $this->assertTrue($findologicProduct->hasAttributes());
+        $attributes = $findologicProduct->getAttributes();
+        if (count($expectedCatUrls) > 0) {
+            $catValues = $attributes[1]->getValues();
+        } else {
+            $catValues = $attributes[0]->getValues();
+        }
+        foreach ($catValues as $category) {
+            if (!empty(array_search($category, $categoriesToBeChecked))) {
+                $isUnique = false;
+                break;
+            }
+            $categoriesArray[] = $category;
+        }
+        $this->assertTrue($isUnique);
+    }
+
+    public function testSetVariantCategoryOnly()
```

Comment:
@D3xter7 What is this test doing? There is nothing being asserted here. Please make some assertion that tell us what categories are returned.

Reply:
I only needed to check if findologic product is being exported.

Reply:
@D3xter7 Click-test is something different than Unit Tests. In click-test you need to ensure that the product is exported, but here you are writing unit tests, so it should test the change which you introduced, which in this case is the categories. Please check the test case section on Youtrack it is clearly mentioned to ensure the configured categories are exported properly.